PhysicsTools/PatAlgos/plugins/PATPackedCandidateProducer.cc

```diff
@@ -29,431 +29,451 @@
-            outPtrP->back().setMuonID(cand.muonRef()->isStandAloneMuon(), cand.muonRef()->isGlobalMuon());
-          }
+      outPtrP->push_back(
+          pat::PackedCandidate(cand.polarP4(), vtx, ptTrk, etaAtVtx, phiAtVtx, cand.pdgId(), PVRefProd, PV.key()));
+      outPtrP->back().setAssociationQuality(pat::PackedCandidate::PVAssociationQuality(qualityMap[quality]));
+      outPtrP->back().setCovarianceVersion(covarianceVersion_);
+      if (cand.trackRef().isNonnull() && PVOrig.isNonnull() && PVOrig->trackWeight(cand.trackRef()) > 0.5 &&
+          quality == 7) {
+        outPtrP->back().setAssociationQuality(pat::PackedCandidate::UsedInFitTight);
+      }
+      // properties of the best track
+      outPtrP->back().setLostInnerHits(lostHits);
+      if (outPtrP->back().pt() > minPtForTrackProperties_ || outPtrP->back().ptTrk() > minPtForTrackProperties_ ||
+          whiteList.find(ic) != whiteList.end() ||
+          (cand.trackRef().isNonnull() && whiteListTk.find(cand.trackRef()) != whiteListTk.end())) {
+        outPtrP->back().setFirstHit(ctrack->hitPattern().getHitPattern(reco::HitPattern::TRACK_HITS, 0));
+        if (abs(outPtrP->back().pdgId()) == 22) {
+          outPtrP->back().setTrackProperties(*ctrack, covariancePackingSchemas_[4], covarianceVersion_);
         } else {
-
-          if (!PVs->empty()) {
-            PV = reco::VertexRef(PVs, 0);
-            PVpos = PV->position();
+          if (ctrack->hitPattern().numberOfValidPixelHits() > 0) {
+            outPtrP->back().setTrackProperties(
+                *ctrack, covariancePackingSchemas_[0], covarianceVersion_);  //high quality
+          } else {
+            outPtrP->back().setTrackProperties(*ctrack, covariancePackingSchemas_[1], covarianceVersion_);
           }
-	
-          outPtrP->push_back( pat::PackedCandidate(cand.polarP4(), PVpos, cand.pt(), cand.eta(), cand.phi(), cand.pdgId(), PVRefProd, PV.key()));
-          outPtrP->back().setAssociationQuality(pat::PackedCandidate::PVAssociationQuality(pat::PackedCandidate::UsedInFitTight));
         }
-    
-	// neutrals and isolated charged hadrons
-
-        bool isIsolatedChargedHadron = false;
-        if(storeChargedHadronIsolation_) {
-          const edm::ValueMap<bool>  &  chargedHadronIsolation=*(chargedHadronIsolationHandle.product());
-          isIsolatedChargedHadron=((cand.pt()>minPtForChargedHadronProperties_)&&(chargedHadronIsolation[reco::PFCandidateRef(cands,ic)]));
-          outPtrP->back().setIsIsolatedChargedHadron(isIsolatedChargedHadron);
+        //outPtrP->back().setTrackProperties(*ctrack,tsos.curvilinearError());
+      } else {
+        if (outPtrP->back().pt() > 0.5) {
+          if (ctrack->hitPattern().numberOfValidPixelHits() > 0)
+            outPtrP->back().setTrackProperties(
+                *ctrack, covariancePackingSchemas_[2], covarianceVersion_);  //low quality, with pixels
+          else
+            outPtrP->back().setTrackProperties(
```

Comment:
@davidlange6 @fabiocos 
is there a way to add braces in clang-format for the cases of multi-line break-up of an if/else?
The result looks like a bit more dangerous pattern where and else can get misplaced

Reply:
no - clang-format changes whitespace only. There is a clang-tidy option to add braces for ifs/fors/whiles  that has been tried but so far doesn't work fully in cmssw (afaik).